app/bundles/PluginBundle/Command/FetchLeadsCommand.php

```diff
@@ -105,17 +117,28 @@ protected function execute(InputInterface $input, OutputInterface $output)
             $params['start'] = $startDate;
             $params['end']   = $endDate;
             $params['limit'] = $limit;
+
             if (isset($supportedFeatures) && in_array('get_leads', $supportedFeatures)) {
                 if ($integrationObject !== null && method_exists($integrationObject, 'getLeads') && isset($config['objects'])) {
                     $output->writeln('<info>'.$translator->trans('mautic.plugin.command.fetch.leads', ['%integration%' => $integration]).'</info>');
-                    if (strtotime($startDate) > strtotime('-30 days')) {
-                        if (in_array('Lead', $config['objects'])) {
-                            $processed = intval($integrationObject->getLeads($params, null, $leads, [], 'Lead'));
+                    if (strtotime($startDate) > strtotime('-30 days') || $input->getOption('fetch-all')) {
+
+                        //Handle case when integration object are named "Contacts" and "Leads"
+                        $lead_object_name = 'Lead';
```

Comment:
I really dislike this, as it promotes code bloat. Every new integration we add, following this pattern, will have to add lines to this class in order to fetch leads and companies. I propose adding a method like the following to be added to the `CrmAbstractIntegrationClass`:

```php
    protected $objectNameMap   = [];
    //...
    /**
     * Returns the integration's name for a Mautic object.
     *
     * @param string $key
     *
     * @return string|null
     */
    public function getObjectName($key)
    {
        // The baseObjectNameMap is structured as this to retain BC with current integration naming
        $baseObjectNameMap = [
            'contact' => 'Contact',
            'company' => 'company',
        ];

        $map = array_merge($baseObjectNameMap, $this->objectNameMap);

        if (array_key_exists($key, $map)) {
            return $map[$key];
        }

        return null;
    }
```

Then, in each class, you use the `$objectNameMap` property to map them to what they are supposed to be. The keys represent what Mautic calls them; the values represent what the CRM calls them. 

```php

    protected $objectNameMap   = [
        'contact' => 'Lead'
    ];
```

Usage with this will be:

```php
$contactObjectName = $integrationObject->getObjectName('contact');
$companyObjectName = $integrationObject->getObjectName('company');
//...
if (in_array($contactObjectName, $config['objects']) {
    $processed += intval($integrationObject->getLeads($params, null, $contact, [], $contactObjectName);
}
```